File: toys_raffle/presenters/Presenter.java
package toys_raffle.presenters;

import java.util.Map;

import toys_raffle.commands.Command;
import toys_raffle.views.Msg;
import toys_raffle.views.View;

/**
 * Presenter class of MVP
 */
public class Presenter {
    private View view;
    private Map<Integer, Command> commands;

    /**
     * Default constructor for Presenter instance
     * @param view      View implementation
     * @param commands  Commands map
     */
    public Presenter(View view, Map<Integer,Command> commands) {
        this.view = view;
        this.commands = commands;
    }
    
    /**
     * Runs the application presentation
     */
    public void Run() {
        this.view.ShowMsg(Msg.welcomeMsg);
        int cmd = -1;

        while (cmd != 0) {
            this.view.MainMenu();
            cmd = this.view.getInt(Msg.getIndexMsg);
            if (this.commands.containsKey(cmd)) {
                this.commands.get(cmd).execute();
            } else {
                this.view.ShowMsg(Msg.notImplementedMsg);
            }
        }
    }
}

File: notes-python/presenter.py
import msg
from view import View
from commands import Invoker

DEFAULT_CMD_NUMBER = -1
EXIT_CMD_NUMBER = 0

class Presenter:
    """Class for Presenter instances

    Attributes
    ----------
    __view : View
        View instance to interact with user
    __invoker : Invoker
        Invoker instance to execute commands
    """
    
    def __init__(self, view:View, invoker:Invoker) -> None:
        """Default constructor for Notes instance
        
        Parameters:
            view: View instance to interact with user (View)
            invoker: Invoker instance to execute commands (Invoker)
        """
        
        self.__view = view
        self.__invoker = invoker

    def run(self) -> None:
        """Run application using Presenter instance"""
        
        self.__view.print_msg(msg.WELCOME_MSG)

        cmd_number = DEFAULT_CMD_NUMBER
        while cmd_number != EXIT_CMD_NUMBER:
            self.main_menu()
            cmd_number = self.__view.get_int(msg.GET_COMMAND_MSG)
            cmd = self.__invoker.get_command(cmd_number)
            if cmd is not None:
                cmd.execute()

        self.__view.print_msg(msg.BYE_MSG)
    
    def main_menu(self) -> None:
        """Show Main menu"""

        self.__view.print_msg(msg.COMMANDS_MENU_MSG)
        commands = self.__invoker.get_commands()
        for key, value in commands.items():
            self.__view.print_msg(f"{key} - {value}")
